src/workerd/api/node/crypto_util.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <cstdio>
+
+// TODO: Move to general node/util.h
+#ifdef __GNUC__
+#define MUST_USE_RESULT __attribute__((warn_unused_result))
+#else
+#define MUST_USE_RESULT
+#endif
+
+// TODO:
+namespace workerd::api::node::CryptoUtil {
+
+struct CSPRNGResult {
+  const bool ok;
+  MUST_USE_RESULT bool is_ok() const { return ok; }
+  MUST_USE_RESULT bool is_err() const { return !ok; }
+};
+
+// Either succeeds with exactly |length| bytes of cryptographically
+// strong pseudo-random data, or fails. This function may block.
+// Don't assume anything about the contents of |buffer| on error.
+// As a special case, |length == 0| can be used to check if the CSPRNG
+// is properly seeded without consuming entropy.
+MUST_USE_RESULT CSPRNGResult CSPRNG(void* buffer, size_t length);
```

Comment:
I'm not yet convinced we should port this vs. using the entropy source that is already in workerd.  A key difference between the two is that the existing workerd implementation does not allow generating random values outside of an IoContext, which means it cannot be used at the global scope. I'd be interested in what @kentonv thinks here.